#! /usr/bin/env python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("red_histogram_bin", int_t,    0, "the size of the bin for red", 1, 10, 256)
gen.add("green_histogram_bin", int_t,    0, "the size of the bin for green", 1, 10, 256)
gen.add("blue_histogram_bin", int_t,    0, "the size of the bin for blue", 1, 10, 256)
gen.add("hue_histogram_bin", int_t,    0, "the size of the bin for hue", 1, 10, 180)
gen.add("saturation_histogram_bin", int_t,    0, "the size of the bin for saturatoin", 1, 10, 256)
gen.add("intensity_histogram_bin", int_t,    0, "the size of the bin for intensity", 1, 10, 256)

exit(gen.generate(PACKAGE, "color_histogram", "ColorHistogram"))
